In [7]:
import pandas as pd
from collections import defaultdict
 
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

#A case
val = df.Survived.value_counts(normalize=True)
print(val)

#B Case
bVal = df.groupby('Sex')['Survived'].mean()
print(bVal)

#C Case
Q1 = df['Fare'].quantile(0.25)
Q3 = df['Fare'].quantile(0.75)

iqr = Q3 - Q1

max = Q3 + (1.5 * iqr)

print(max)
countMax = 0
map_val = defaultdict(int)

for val in df.itertuples():
    if val.Fare > max:
        countMax += 1
        map_val[val.Pclass] += 1

print("Count High : " , countMax)
print("Map : ", map_val)

#D Case
avg = df.Fare.mean()
median = df.Fare.median()

print("Average : ", avg)
print("Median : ", median)


Survived
0    0.616162
1    0.383838
Name: proportion, dtype: float64
Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64
65.6344
Count High :  116
Map :  defaultdict(<class 'int'>, {1: 104, 2: 5, 3: 7})
Average :  32.204207968574636
Median :  14.4542


Pregunta A (Sumarización Categórica): Usen la función .value_counts(normalize=True) en la columna Survived (0 = Murió, 1 = Sobrevivió). ¿Cuál es la tasa de supervivencia global del barco expresada en porcentaje?

De acuerdo a la funcion, el 38% de las personas del barco sobrevivieron al accidente

Pregunta B : El famoso código marítimo era "mujeres y niños primero". Comprobemos esto matemáticamente. Ejecuten una agrupación por la columna de género y calculen el promedio de la columna Survived. ¿Qué porcentaje exacto de mujeres sobrevivió en contraste con los hombres?

El 74% de mujeres sobrevivieron al accidente en contraste al 18% de hombres

Pregunta C: El director financiero nota que la tarifa máxima (Fare) cobrada fue de más de 500 libras, mientras que el promedio ronda las 32. Sospecha de outliers. Utilicen las herramientas de dispersión en Pandas para confirmarlo:
Calculen el Cuartil 1 (25%) y el Cuartil 3 (75%) de la columna Fare usando el método .quantile([0.25, 0.75]).
Calculen matemáticamente el Rango Intercuartílico (IQR = Q3 - Q1).
Calculen el límite superior aceptable (Q3 + 1.5 * IQR). Escriban una línea de código para filtrar el DataFrame y descubrir exactamente cuántos pasajeros pagaron una tarifa por encima de ese límite matemático. ¿A qué clase (Pclass) pertenecían la mayoría de ellos?

Fueron 116 pasajeros que pagaron una tarifa mayor y 104 de ellos eran de la clase 1

Pregunta D: Calculen la media y la mediana de la columna Fare. Notarán una diferencia enorme entre ambos valores. Matemáticamente, ¿qué significa que la media sea tan superior a la mediana? Si en el futuro utilizamos un algoritmo basado en distancias euclidianas (como K-Nearest Neighbors) sin escalar previamente esta variable, ¿cómo afectará esta asimetría al aprendizaje del modelo?
1. Significa que hay muchos 